### ISSUE TITLE
Version 0.4.0 Yanked

### ISSUE THREAD

**Issue by NJAldwin** [hwchen/keyring-rs#11]
I see that version 0.4.0 has recently been yanked from crates.io, but I don't see anything stating why.  Was there a security vulnerability?  0.3.0 hasn't been yanked; is it still safe to use?

**Comment by hwchen** [comment]
Sorry, I've been a little behind in updating this library.

The reason I yanked 0.4.0 was because I was going to update the dependency to my secret-service library. Unfortunately, the bug I fixed was not actually fixed (turned out to be unfixable because of a bug encoding empty strings in rust-crypto), so I yanked that version of ss, and then also keyring. I waited a bit to think about what I should do about it, but then got caught up in other things.

On the plus side, the yank was not because of security vulnerabilities. It just means there's been inconsistent handling of empty strings as input across different OS. (something which will continue until rust-crypto gets updated).

Thanks for this reminder, though. Other updates have been made to both keyring and secret-service which warrant publishing, which I'll do shortly.
